gating/generate_release_notes/run

```diff
@@ -14,4 +14,20 @@
 # RE_HOOK_RELEASE_NOTES: The file where release notes should be written. The
 # release notes should be formatted with GitHub-flavoured Markdown.
 # RE_HOOK_REPO_HTTP_URL: The URL for the repository being released e.g.
-# https://github.com/rcbops/rpc-openstack.git.
\ No newline at end of file
+# https://github.com/rcbops/rpc-openstack.git.
+
+# This script is called by Jenkins to generate release notes.
+
+# The resulting notes must be written to $WORKSPACE/artifacts/release_notes
+
+export PREVIOUS_TAG=${RE_HOOK_PREVIOUS_VERSION}
+export NEW_TAG=${RE_HOOK_VERSION}
+export REPO_URL=${RE_HOOK_REPO_HTTP_URL}
+
+apt-get install -y pandoc
+
+pip install reno==2.5.1
+
+gating/generate_release_notes/generate_release_notes.sh
+
+cp all_notes.md "${RE_HOOK_RELEASE_NOTES}"
```

Comment:
we have a lot of error handling in the called scripts (ie generate_release_notes.sh, generate_reno_report.sh), but none here, though i would suggest:
* `set -e` (because if pandoc or reno installation fails, no use in continuing only to fail later and with a potentially confusing error)
* `if generate_release_notes.sh; then cp ...; else echo Release note generation failed; fi`